src/inmanta/plugins.py

```diff
@@ -841,13 +842,31 @@ def type_string(self) -> str:
         return self.get_full_name()
 
 
+@typing.overload
+def plugin(
+    function: str | None = None,
```

Comment:
Nice, the change is even more meaningful here, with the `str | Callable | None`!